vector_helpers.go

```diff
@@ -0,0 +1,82 @@
+package cimgui
+
+// implement vector operations
+
+// Vec2:
+
+// Add returns a sum of v and another.
+func (v ImVec2) Add(another ImVec2) ImVec2 {
+	return ImVec2{
+		X: v.X + another.X,
+		Y: v.Y + another.Y,
+	}
+}
+
+// Sub returns the vector v - another.
+func (v ImVec2) Sub(another ImVec2) ImVec2 {
+	return ImVec2{
+		X: v.X - another.X,
+		Y: v.Y - another.Y,
+	}
+}
+
+// Mul returns the vector v*k.
+func (v ImVec2) Mul(k float32) ImVec2 {
+	return ImVec2{
+		X: v.X * k,
+		Y: v.Y * k,
+	}
+}
+
+// Div returns the vector v/k.
+func (v ImVec2) Div(k float32) ImVec2 {
+	return ImVec2{
+		X: v.X / k,
+		Y: v.Y / k,
+	}
+}
+
+// Eq reports whether p and q are equal.
+func (v ImVec2) Eq(q ImVec2) bool {
```

Comment:
This is not needed. You can just do `a == b` for two vectors.

Reply:
In fact, it isn't equivalent of `==` operation.
it calls `Canon()` to invert both rects (`ImVec4`s) to canonical form.
This method for example will return `true` for
```go
ImVec4{-4, -4, 4, 4}.Eq(ImVec4{4, 4, -4, -4})
```

it follows idea of `image.Rectangle`

Reply:
We're talking about Vec2 here.

Reply:
oh, sorry :smile:
so yah, if two `vec2`s can be compared with `==`, this one can be removed

Reply:
well, so removing